docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+        To check a specific instance, run:
+        edgedb -I <name> migration log --from-db --newest-first --limit 1
+    1. Merge version control branches that contain schema changes
+    if possible.
+
+    Proceed? [y/n]
+
+Then just press ``y`` to squash all of your existing migrations into 
+a single file.
+
+Fixups during a squash
+----------------------
+
+If your schema doesn't match the schema in the database, EdgeDB will 
+prompt you to create a *fixup* file, which can be useful (as the CLI 
+says) to "automate upgrading other instances to a squashed revision". 
+You'll see fixups inside a folder called ``/fixups``. Their file names 
+are extremely long, but are simply two migration hashes joined together 
+by a dash. So a fixup that begins with
+
+.. code-block::
+
+    CREATE MIGRATION 
+    m1v3vqmwif4ml3ucbzi555mjgm4myxs2husqemopo2sz2m7otr22ka 
+    ONTO m16awk2tzhtbupjrzoc4fikgw5okxpfnaazupb6rxudxwin2qfgy5q
+
+will have a file name a full 116 characters in length.
+
+The CLI output when using squash along with a fixup is pretty informative 
+on its own, so let's just walk through the output as you'll see it 
+in practice.
+
+First we'll begin with this schema:
+
+.. code-block:: sdl
+
+  type User {
+    name: str;
+  }
+
+Then remove ``name: str;`` from the ``User`` type, migrate, put it 
+back again, and migrate. You'll now have multiple files in your
+``/migrations`` folder and will now be able to squash them.
+
+Then change to this schema without migrating it:
+
+.. code-block:: sdl
+
+  type User {
+    name: str;
+    nickname: str;
+  }
+
+Upon using ``edgedb migration create --squash``, the output is first 
+the same as with our previous squash:
+
+.. code-block:: bash
+
+    $ edgedb migration create --squash
+    Current database revision: 
+    m16awk2tzhtbupjrzoc4fikgw5okxpfnaazupb6rxudxwin2qfgy5q
+    While squashing migrations is non-destructive, 
+    it may lead to manual work if done incorrectly.
+
+    Items to check before using --squash:
+    1. Ensure that `./dbschema` dir is comitted
+    2. Ensure that other users of the database have the revision
+    above or can create database from scratch.
+        To check a specific instance, run:
+        edgedb -I <name> migration log --from-db --newest-first --limit 1
+    3. Merge version control branches that contain schema changes 
+    if possible.
+
+    Proceed? [y/n]
+    > y
+
+But after typing ``y``, the CLI will notice that the existing schema 
+differs from what you have and offers to make a fixup file:
+
+.. code-block::
+
+    Your schema differs from the last revision. 
+    A fixup file can be created
+    to automate upgrading other instances to a squashed revision.
+    This starts the usual migration creation process.
+
+    Feel free to skip this step if you don't have
+    other instances to migrate
+
+    Create a fixup file? [y/n]
+    > y
+
+You will then see the the same questions that otherwise show up in 
```

Comment:
```suggestion
You will then see the the same questions that otherwise show up in 
```
```suggestion
You will then see the the same questions that would otherwise show up in 
```